### PR TITLE
DEV-4033 Implement CompTextConstructor 

### PR DESCRIPTION
**Pull request by crodriguezq**

### CODE REVIEW

File: src/expressions/xquery/CompTextConstructor.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	constructor(expr: Expression | null) {
+		super(expr ? expr.specificity : new Specificity({}), expr ? [expr] : [], {
+			canBeStaticallyEvaluated: false,
+			resultOrder: RESULT_ORDERINGS.UNSORTED
+		});
+
+		this._expr = expr;
+	}
+
+	public evaluate(_dynamicContext, executionParameters) {
```
Comment (JosVerburg) [inline src/expressions/xquery/CompTextConstructor.ts]: Lets add typing for these. `public evaluate(_dynamicContext: DynamicContext, executionParameters: ExecutionParameters) {`

Comment (JosVerburg) [inline src/expressions/xquery/CompTextConstructor.ts]: `_dynamicContext` is used so it shouldn't have the `_` prefix.

File: src/parsing/compileAstToExpression.ts [hunk 2]
```diff
@@ -1064,6 +1067,17 @@ function computedCommentConstructor(ast, compilationOptions) {
 	return new CommentConstructor(expr);
 }
 
+function computedTextConstructor(ast, compilationOptions) {
```
Comment (JosVerburg) [inline src/parsing/compileAstToExpression.ts]: No need to type everything is this file, but I think we should make sure everything new is typed.
`function computedTextConstructor(ast: IAST, compilationOptions: CompilationOptions) {`

File: src/expressions/xquery/CompTextConstructor.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import Expression, { RESULT_ORDERINGS } from '../Expression';
+import Specificity from '../Specificity';
+
+import castToType from '../dataTypes/castToType';
+import createNodeValue from '../dataTypes/createNodeValue';
+import sequenceFactory from '../dataTypes/sequenceFactory';
+
+class CompTextConstructor extends Expression {
```
Comment (JosVerburg) [inline src/expressions/xquery/CompTextConstructor.ts]: I suggest to remove the `Comp` prefix as we also do not use this for the other computed constructors.

**Comment by crodriguezq** [comment]
Thanks for your comments @JosVerburg

**Pull request merged**